### PR TITLE
[action] Improve get_version_number: fallback to project settings if no MARKETING_VERSION is set for target

### PR DESCRIPTION
**Pull request by ilendemli**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
https://github.com/fastlane/fastlane/issues/15322

### Description
`get_version_number_from_build_settings` now returns `nil` if no version number is found instead of throwing an error
the `run` method will throw the error if version_number is not set
it will look into the project settings for `variable` (in this case `MARKETING_VERSION`) if there is no `version_number` set after checking the target

### Testing Steps
`bundle exec fastlane store` resulted with the correct value after these changes

### CODE REVIEW

**Comment by ilendemli** [comment]
@googlebot I signed it!

**Comment by michael-patzer** [comment]
This is an important improvement for us, and a use-case that I can see being common is setting the MARKETING_VERSION on the project and having all of the targets inherit that value with $(MARKETING_VERSION) to keep versions aligned across multiple related targets (app, watch, extensions). Thank you for this addition.

File: fastlane/lib/fastlane/actions/get_version_number.rb [hunk 1]
```diff
@@ -26,6 +26,16 @@ def self.run(params)
           version_number = get_version_number_from_build_settings!(target, $1, configuration)
         end
 
+        # Check if version_number is not set and try retrieving value from project settings
+        if version_number.nil?
+          version_number = get_version_number_from_build_settings!(project, $1, configuration)
```
Comment (joshdholtz) [inline fastlane/lib/fastlane/actions/get_version_number.rb]: This usage of $1 here seems to be a bit confusing 🤔 It isn't really tied to any specific regex here. I think this fallback almost needs to go inside the if statements above. Thoughts?

**Review by joshdholtz — CHANGES_REQUESTED** [review]
I like the change but I think things just need to be moved around 😊

File: fastlane/lib/fastlane/actions/get_version_number.rb (hunk 1 above)
Comment (ilendemli) [reply fastlane/lib/fastlane/actions/get_version_number.rb]: you are absolutely correct, the downside would be having the same line twice, one in each if check

**Comment by michael-patzer** [comment]
@ilendemli Looks like one of the CI tests failed (Xcode 9.0.1, Ruby 2.3). Bringing to your attention in case you hadn't seen it.

**Comment by ilendemli** [comment]
@michael-patzer I have seen it fail, I am just not sure if it my fault that it is failing or if that was just an error that just happened. @joshdholtz Could you check this out? I can't rerun tests.

**Comment by joshdholtz** [comment]
@googlebot I consent.

**Review by joshdholtz — APPROVED** [review]
Thanks for making those requested changes 😊 I added some tests for them and rebased from master. Appreciate that contribution ❤️

**Pull request merged**